File: Models/Tab.cs
using System.ComponentModel.DataAnnotations;
using Maia.Models.Abstracts;
using Maia.Models.DTO;

namespace Maia.Models
{
    public class Tab : AnotacaoMusical
    {

        [MaxLength(90)]
        public string Titulo { get; set; }
        
        [MaxLength(140)]
        public string Descricao { get; set; }
        
        public string Texto { get; set; }
        public string TextoHtml { get; set; }
        
        [MaxLength(8)]
        public string Afinacao { get; set; }
        

        [RegularExpression("Guitarra|Baixo|Violao")]
        public string Instrumento { get; set; } = "Guitarra";
        
        public Tab() { }
        
        public Tab(TabDTO dto)
        {
            this.MusicaId = dto.MusicaId;
            this.UsuarioId = dto.UsuarioId;
            this.Instrumento = dto.Instrumento;
            this.Titulo = dto.Titulo;
            this.Descricao = dto.Descricao;
            this.Texto = dto.Texto;
            this.TextoHtml = dto.TextoHtml;
            this.Afinacao = dto.Afinacao;
        }

    }
}

File: Angular/src/app/shared/models/Tab.ts
import {AnotacaoMusical} from "./AnotacaoMusical";
import {Usuario} from "./Usuario";
import {Musica} from "./Musica";

export type instrumento = 'Baixo' | 'Violao' | 'Guitarra';

export class Tab extends AnotacaoMusical {
  public titulo: string;
  public texto: string;
  public textoHtml: string;
  public afinacao: string;
  public descricao: string;
  public instrumento: instrumento;

  constructor(
    musicaId: number,
    usuarioId: number,
    titulo: string,
    texto: string,
    textoHtml: string,
    afinacao: string,
    descricao: string,
    instrumento: instrumento,
    usuario?: Usuario,
    musica?: Musica,
    id?: number
  ) {
    super(musicaId, usuarioId, usuario, musica, id);
    this.titulo = titulo;
    this.texto = texto;
    this.textoHtml = textoHtml;
    this.afinacao = afinacao;
    this.descricao = descricao;
    this.instrumento = instrumento;
  }
}
